pkg/ui/conflict/conflict.go

```diff
@@ -44,19 +72,125 @@ func Solve(in io.Reader, out io.Writer, filePath string, fileA, fileB []byte) ([
 }
 
 func NewModel(filePath string, fileA, fileB []byte) Model {
-	return Model{
-		filePath: filePath,
-		fileA:    fileA,
-		fileB:    fileB,
+
+	fileDiff := diff.New(string(fileA), string(fileB))
+
+	m := Model{
+		filePath:   filePath,
+		fileA:      fileA,
+		fileB:      fileB,
+		diff:       fileDiff,
+		ready:      false,
+		resolution: UseOld,
 	}
+
+	return m
 }
 
 func (m Model) Init() tea.Cmd {
 	return nil
 }
 
+func calculateViewportHeight(windowHeight, verticalMarginal int) int {
+	viewportHeight := windowHeight - verticalMarginal
+	return viewportHeight
+}
+
+func (m Model) headerView() string {
+	title := titleStyle.Render(m.filePath)
+	line := strings.Repeat("─", max(0, m.viewport.Width-lipgloss.Width(title)))
+	viewPortHeader := lipgloss.JoinHorizontal(lipgloss.Center, title, line)
+	conflictPrompt := fmt.Sprintf("There are conflicts in the following file: %s, what do you want to do?", m.filePath)
+	return fmt.Sprintf("%s\n%s", conflictPrompt, viewPortHeader)
+}
+
+func (m Model) footerView() string {
+	info := infoStyle.Render(fmt.Sprintf("%3.f%%", m.viewport.ScrollPercent()*100))
+	line := strings.Repeat("─", max(0, m.viewport.Width-lipgloss.Width(info)))
+	progressLine := lipgloss.JoinHorizontal(lipgloss.Center, line, info)
+
+	overwriteOpt := "Overwrite old"
+	keepOpt := "Keep old"
+	diffOpt := "Write the file with diffs"
+	var resolutionSelector string
+	if m.resolution == UseOld {
+		resolutionSelector = fmt.Sprintf("> %s / %s / %s", lipgloss.NewStyle().Bold(true).Render(keepOpt), overwriteOpt, diffOpt)
+	} else if m.resolution == UseNew {
+		resolutionSelector = fmt.Sprintf("%s / > %s / %s", keepOpt, lipgloss.NewStyle().Bold(true).Render(overwriteOpt), diffOpt)
+	} else {
+		resolutionSelector = fmt.Sprintf("%s / %s / > %s", keepOpt, overwriteOpt, lipgloss.NewStyle().Bold(true).Render(diffOpt))
+	}
+
+	instructions := "Use up and down arrows to move up and down the diff file.\nPage down to go the the next conflict and page up to go to the previous conflict."
+
+	return lipgloss.JoinVertical(0, progressLine, instructions, resolutionSelector)
+}
+
+func combineDiffLinesToColorizedDiffString(lines []string) string {
+
+	oldStyle := lipgloss.NewStyle().Foreground(lipgloss.Color("#FF0000"))
+	newStyle := lipgloss.NewStyle().Foreground(lipgloss.Color("#00FF00"))
```

Comment:
I wonder is the raw green and red best options here. For example there are ["common" green and red](https://github.com/futurice/jalapeno/blob/main/internal/cli/root.go#L20-L21) used throughout the code

Reply:
I changed them to match these, I am not sure if it made it look better but consistency within the app is nice to have.